### PR TITLE
A litle fix for this plugin, but some bugs contains

### PR DESCRIPTION
**Pull request by supercrafter333**
Contained bugs:
- cant nim beacon out of creativInventory
- Error:
```
[19:55:00] [Server thread/CRITICAL]: Error: "Undefined property: jasonwynn10\beacon\packet\InventoryTransactionPacketV2::$transactionType" (EXCEPTION) in "pmsrc/src/pocketmine/network/mcpe/protocol/DataPacket" at line 205
[19:55:00] [Server thread/CRITICAL]: #0 plugins/PM-BeaconsGH/src/jasonwynn10/beacon/packet/InventoryTransactionPacketV2(46): pocketmine\network\mcpe\protocol\DataPacket->__set(string[15] transactionType, integer 0)
[19:55:00] [Server thread/CRITICAL]: #1 pmsrc/src/pocketmine/network/mcpe/protocol/DataPacket(92): jasonwynn10\beacon\packet\InventoryTransactionPacketV2->decodePayload()
[19:55:00] [Server thread/CRITICAL]: #2 pmsrc/src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(101): pocketmine\network\mcpe\protocol\DataPacket->decode()
[19:55:00] [Server thread/CRITICAL]: #3 pmsrc/src/pocketmine/network/mcpe/protocol/BatchPacket(130): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(object jasonwynn10\beacon\packet\InventoryTransactionPacketV2)
[19:55:00] [Server thread/CRITICAL]: #4 pmsrc/src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(109): pocketmine\network\mcpe\protocol\BatchPacket->handle(object pocketmine\network\mcpe\PlayerNetworkSessionAdapter)
[19:55:00] [Server thread/CRITICAL]: #5 pmsrc/src/pocketmine/Player(3311): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(object pocketmine\network\mcpe\protocol\BatchPacket)
[19:55:00] [Server thread/CRITICAL]: #6 pmsrc/src/pocketmine/network/mcpe/RakLibInterface(169): pocketmine\Player->handleDataPacket(object pocketmine\network\mcpe\protocol\BatchPacket)
[19:55:00] [Server thread/CRITICAL]: #7 pmsrc/vendor/pocketmine/raklib/src/server/ServerHandler(95): pocketmine\network\mcpe\RakLibInterface->handleEncapsulated(string[18] 92.195.61.73 51012, object raklib\protocol\EncapsulatedPacket, integer 0)
[19:55:00] [Server thread/CRITICAL]: #8 pmsrc/src/pocketmine/network/mcpe/RakLibInterface(109): raklib\server\ServerHandler->handlePacket()
[19:55:00] [Server thread/CRITICAL]: #9 pmsrc/src/pocketmine/network/mcpe/RakLibInterface(99): pocketmine\network\mcpe\RakLibInterface->process()
[19:55:00] [Server thread/CRITICAL]: #10 pmsrc/vendor/pocketmine/snooze/src/SleeperHandler(113): pocketmine\network\mcpe\RakLibInterface->pocketmine\network\mcpe\{closure}()
[19:55:00] [Server thread/CRITICAL]: #11 pmsrc/vendor/pocketmine/snooze/src/SleeperHandler(75): pocketmine\snooze\SleeperHandler->processNotifications()
[19:55:00] [Server thread/CRITICAL]: #12 pmsrc/src/pocketmine/Server(2155): pocketmine\snooze\SleeperHandler->sleepUntil(double 1617825300.3895)
[19:55:00] [Server thread/CRITICAL]: #13 pmsrc/src/pocketmine/Server(1992): pocketmine\Server->tickProcessor()
[19:55:00] [Server thread/CRITICAL]: #14 pmsrc/src/pocketmine/Server(1586): pocketmine\Server->start()
[19:55:00] [Server thread/CRITICAL]: #15 pmsrc/src/pocketmine/PocketMine(291): pocketmine\Server->__construct(object BaseClassLoader, object pocketmine\utils\MainLogger, string[11] /home/Dev2/, string[19] /home/Dev2/plugins/)
[19:55:00] [Server thread/CRITICAL]: #16 pmsrc/src/pocketmine/PocketMine(321): pocketmine\server()
[19:55:00] [Server thread/CRITICAL]: #17 pmsrc(11): require(string[63] phar:///home/Dev2/pocketmine.phar/src/pocketmine/PocketMine.php)

```

### CODE REVIEW

**Comment by supercrafter333** [comment]
Fixed the called error

**Comment by supercrafter333** [comment]
New error:
```
[21:19:37] [Server thread/CRITICAL]: UnexpectedValueException: "Unknown inventory action source type 460" (EXCEPTION) in "pmsrc/src/pocketmine/network/mcpe/protocol/types/NetworkInventoryAction" at line 107
[21:19:37] [Server thread/CRITICAL]: #0 plugins/PM-BeaconsGH/src/jasonwynn10/beacon/packet/InventoryTransactionPacketV2(84): pocketmine\network\mcpe\protocol\types\NetworkInventoryAction->read(object jasonwynn10\beacon\packet\InventoryTransactionPacketV2, boolean )
[21:19:37] [Server thread/CRITICAL]: #1 pmsrc/src/pocketmine/network/mcpe/protocol/DataPacket(92): jasonwynn10\beacon\packet\InventoryTransactionPacketV2->decodePayload()
[21:19:37] [Server thread/CRITICAL]: #2 pmsrc/src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(101): pocketmine\network\mcpe\protocol\DataPacket->decode()
[21:19:37] [Server thread/CRITICAL]: #3 pmsrc/src/pocketmine/network/mcpe/protocol/BatchPacket(130): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(object jasonwynn10\beacon\packet\InventoryTransactionPacketV2)
[21:19:37] [Server thread/CRITICAL]: #4 pmsrc/src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(109): pocketmine\network\mcpe\protocol\BatchPacket->handle(object pocketmine\network\mcpe\PlayerNetworkSessionAdapter)
[21:19:37] [Server thread/CRITICAL]: #5 pmsrc/src/pocketmine/Player(3311): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(object pocketmine\network\mcpe\protocol\BatchPacket)
[21:19:37] [Server thread/CRITICAL]: #6 pmsrc/src/pocketmine/network/mcpe/RakLibInterface(169): pocketmine\Player->handleDataPacket(object pocketmine\network\mcpe\protocol\BatchPacket)
[21:19:37] [Server thread/CRITICAL]: #7 pmsrc/vendor/pocketmine/raklib/src/server/ServerHandler(95): pocketmine\network\mcpe\RakLibInterface->handleEncapsulated(string[18] 92.195.61.73 52173, object raklib\protocol\EncapsulatedPacket, integer 0)
[21:19:37] [Server thread/CRITICAL]: #8 pmsrc/src/pocketmine/network/mcpe/RakLibInterface(109): raklib\server\ServerHandler->handlePacket()
[21:19:37] [Server thread/CRITICAL]: #9 pmsrc/src/pocketmine/network/mcpe/RakLibInterface(99): pocketmine\network\mcpe\RakLibInterface->process()
[21:19:37] [Server thread/CRITICAL]: #10 pmsrc/vendor/pocketmine/snooze/src/SleeperHandler(113): pocketmine\network\mcpe\RakLibInterface->pocketmine\network\mcpe\{closure}()
[21:19:37] [Server thread/CRITICAL]: #11 pmsrc/vendor/pocketmine/snooze/src/SleeperHandler(75): pocketmine\snooze\SleeperHandler->processNotifications()
[21:19:37] [Server thread/CRITICAL]: #12 pmsrc/src/pocketmine/Server(2155): pocketmine\snooze\SleeperHandler->sleepUntil(double 1617830377.0167)
[21:19:37] [Server thread/CRITICAL]: #13 pmsrc/src/pocketmine/Server(1992): pocketmine\Server->tickProcessor()
[21:19:37] [Server thread/CRITICAL]: #14 pmsrc/src/pocketmine/Server(1586): pocketmine\Server->start()
[21:19:37] [Server thread/CRITICAL]: #15 pmsrc/src/pocketmine/PocketMine(291): pocketmine\Server->__construct(object BaseClassLoader, object pocketmine\utils\MainLogger, string[11] /home/Dev2/, string[19] /home/Dev2/plugins/)
[21:19:37] [Server thread/CRITICAL]: #16 pmsrc/src/pocketmine/PocketMine(321): pocketmine\server()
[21:19:37] [Server thread/CRITICAL]: #17 pmsrc(11): require(string[63] phar:///home/Dev2/pocketmine.phar/src/pocketmine/PocketMine.php)

```

File: plugin.yml [hunk 1]
```diff
@@ -11,3 +11,4 @@ mcpe-protocol:
  - 419
  - 422
  - 428
+ - 431
```
Comment (jasonw4331) [inline plugin.yml:8]: remove the old protocols if they are no longer supported in the current version

File: plugin.yml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 name: PM-Beacons
 main: jasonwynn10\beacon\Beacons
-version: 1.0.8
+version: 1.0.9
 api:
- - 3.0.0
+ - 3.19.0
```
Comment (jasonw4331) [inline plugin.yml]: do not change the API version requirement

File: src/jasonwynn10/beacon/packet/InventoryTransactionPacketV2.php [hunk 3]
```diff
@@ -2,13 +2,73 @@
 declare(strict_types=1);
 namespace jasonwynn10\beacon\packet;
 
+use pocketmine\network\mcpe\NetworkSession as PacketHandlerInterface;
+//use pocketmine\network\mcpe\protocol\types\inventory\InventoryTransactionChangedSlotsHack;
```
Comment (jasonw4331) [inline src/jasonwynn10/beacon/packet/InventoryTransactionPacketV2.php]: please remove useless comments

File: src/jasonwynn10/beacon/packet/InventoryTransactionPacketV2.php [hunk 4]
```diff
@@ -2,13 +2,73 @@
...
+use pocketmine\network\mcpe\protocol\types\inventory\NormalTransactionData;
+use pocketmine\network\mcpe\protocol\types\inventory\ReleaseItemTransactionData;
+use pocketmine\network\mcpe\protocol\types\inventory\TransactionData;
+use pocketmine\network\mcpe\protocol\types\inventory\UseItemOnEntityTransactionData;
+use pocketmine\network\mcpe\protocol\types\inventory\UseItemTransactionData;
+use UnexpectedValueException as PacketDecodeException;
 use jasonwynn10\beacon\inventory\action\CustomInventoryAction;
 use pocketmine\network\mcpe\protocol\InventoryTransactionPacket;
 use pocketmine\network\mcpe\protocol\types\inventory\InventoryTransactionChangedSlotsHack;
+use function count;
```
Comment (jasonw4331) [inline src/jasonwynn10/beacon/packet/InventoryTransactionPacketV2.php]: I don't import functions in my code style. Please remove the import

File: src/jasonwynn10/beacon/packet/InventoryTransactionPacketV2.php [hunk 5]
```diff
@@ -2,13 +2,73 @@
...
+use pocketmine\network\mcpe\NetworkSession as PacketHandlerInterface;
+//use pocketmine\network\mcpe\protocol\types\inventory\InventoryTransactionChangedSlotsHack;
+use pocketmine\network\mcpe\protocol\ProtocolInfo;
+use pocketmine\network\mcpe\protocol\types\inventory\MismatchTransactionData;
+use pocketmine\network\mcpe\protocol\types\inventory\NormalTransactionData;
+use pocketmine\network\mcpe\protocol\types\inventory\ReleaseItemTransactionData;
+use pocketmine\network\mcpe\protocol\types\inventory\TransactionData;
+use pocketmine\network\mcpe\protocol\types\inventory\UseItemOnEntityTransactionData;
+use pocketmine\network\mcpe\protocol\types\inventory\UseItemTransactionData;
+use UnexpectedValueException as PacketDecodeException;
```
Comment (jasonw4331) [inline src/jasonwynn10/beacon/packet/InventoryTransactionPacketV2.php:12]: does PocketMine's class do this too?

**Review by jasonw4331 — CHANGES_REQUESTED** [review]
Please remove useless comments

File: plugin.yml (hunk 1 above)
Comment (supercrafter333) [reply plugin.yml:8]: Okay

**Comment by supercrafter333** [comment]
I can't do more, I do not have the know how to do more. You have to finish it ._.

File: src/jasonwynn10/beacon/packet/InventoryTransactionPacketV2.php (hunk 5 above)
Comment (supercrafter333) [reply src/jasonwynn10/beacon/packet/InventoryTransactionPacketV2.php:12]: I think yes xD

File: plugin.yml (hunk 2 above)
Comment (supercrafter333) [reply plugin.yml]: Support for older versions os no longer given with this changnes...

Comment (jasonw4331) [reply plugin.yml]: no, thats what the protocol is for

**Comment by supercrafter333** [comment]
Please Finish it, I can' t do more.

**Pull request closed without merging**